internal/app/init/main.go

```diff
@@ -31,23 +33,40 @@ func initram() (err error) {
 		return err
 	}
 
-	// Perform the equivalent of switch_root.
+	// Mount the rootfs.
 	log.Println("mounting the rootfs")
-	if err = initializer.Rootfs(); err != nil {
+	mountpoints, err = squashfs.MountPoints(constants.NewRoot)
+	if err != nil {
 		return err
 	}
+	squashfs := manager.NewManager(mountpoints)
+	squashfs.MountAll()
 
-	// Perform the equivalent of switch_root.
+	// Switch into the new rootfs.
 	log.Println("entering the rootfs")
-	if err = initializer.Switch(); err != nil {
+	if err = switchroot.Switch(constants.NewRoot, virtual); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func recovery() {
+	if r := recover(); r != nil {
```

Comment:
not related to this PR, but I remember we had a disk sync handler on the return path from `main()` (probably more for machined, but still), do we need it still or it was removed on purpose?

Reply:
At this point `init` is only the initramfs and no disks have been mounted, so I _think_ we can skip that step. That being said it doesn't hurt to call `sync`.

Reply:
I don't see it in machined either. I can add it back if you don't mind

Reply:
Its handled in a task now: 
https://github.com/talos-systems/talos/blob/master/internal/app/machined/internal/phase/api/api.go#L68
https://github.com/talos-systems/talos/blob/master/internal/app/machined/internal/phase/api/acpi.go#L41

Reply:
oh, I missed that!

I noticed a data race in phase runner, I can send  a PR to fix it.

I also thought we might try to unify that though with any other panic-like behavior to get just a single exit path. I can try to do something later on it.

Reply:
+1 on both!